sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/RuleExecutor.scala

```diff
@@ -44,6 +44,71 @@ object RuleExecutor {
   }
 }
 
+class PlanChangeLogger[TreeType <: TreeNode[_]] extends Logging {
+
+  private val logLevel = SQLConf.get.optimizerPlanChangeLogLevel
+
+  private val logRules = SQLConf.get.optimizerPlanChangeRules.map(Utils.stringToSeq)
+
+  private val logBatches = SQLConf.get.optimizerPlanChangeBatches.map(Utils.stringToSeq)
```

Comment:
shall we rename the config names? just remove `optimizer`.

Reply:
Ah, okay.